### DISCUSSION TITLE
How to do Internationalized Routing with double subpath: country/language ?

### DISCUSSION THREAD

**Discussion by ermakov95** [vercel/next.js#42622] [Help]
Example:
website.com
website.com/en
website.com/tr
website.com/fr
website.com/turkey
website.com/turkey/tr
website.com/turkey/en
website.com/turkey/fr

My next.config.js:
`
  const nextConfig = {
    pageExtensions: ["page.tsx", "page.ts", "page.jsx", "page.js"],
    i18n: {
      locales: ["en", "fr", "tr", "turkey", "turkey/en", "turkey/fr"],
      defaultLocale: "en",
      domains: [
        {
          domain: "http://localhost:3000",
          defaultLocale: "en",
          locales: ["tr", "fr", "turkey", "turkey/en", "turkey/fr"],
        },
      ],
      localeDetection: false,
    },
    async rewrites() {
      return [
        {
          source: "/turkey/en/:path*",
          destination: "/:path*",
          locale: false,
        },
        {
          source: "/turkey/fr/:path*",
          destination: "/fr/:path*",
          locale: false,
        },
        {
          source: "/turkey/:path*",
          destination: "/tr/:path*",
          locale: false,
        },
      ];
    },
  };
    
    module.exports = {
      ...nextConfig,
      webpack(config) {
        config.module.rules.push({
          test: /\.svg$/i,
          issuer: /\.[jt]sx?$/,
          use: ["@svgr/webpack"],
        });
        return config;
      },
    };
`

But locales "turkey/en" and "turkey/fr" just dont work
And rewrites change url in brawser. Example:  http://localhost:3000/turkey/fr/about -> http://localhost:3000/fr/about

**Comment by michalpulpan** [comment]
Did you figure this out? I'm dealing with similar issue - trying to implement `{locale}/{country}` into URL in Next.
But I don't think generating redundant "locales" is the correct way since Next will generate static pages for both `en` and `turkey/en` which will share the same translations and be redundant.

**Reply by ermakov95** [reply]
Unfortunately, no.

**Comment by GabenGar** [comment]
Just stick to using a single path segment with a [BCP-47 tag](https://en.wikipedia.org/wiki/IETF_language_tag) to denote meme locales instead of reinventing your own wheel. Will spare you a lot of headaches down the line and improve your SEO rating to boot, since the search crawler wouldn't have to spend time guessing potential dupes.

**Reply by michalpulpan** [reply]
@GabenGar Yeah, that makes sense + it generates dozens of redundant routes. Do you have a suggestion how to incorporate country into a path segment? Would it be better to denote country as a separate url segment?

**Reply by GabenGar** [reply]
Is country of importance to the locale? BCP-47 can include various ISO country codes, so it will still be within a single path segment, but perhaps too annoying to manage and develop. So better to start off with [basic subtags](https://en.wikipedia.org/wiki/IETF_language_tag#List_of_common_primary_language_subtags) and then slap on more verbose ones when the need arises.

**Reply by ermakov95** [reply]
@GabenGar  Yes, since the question says about "country/language", it means that the country should be in the url.  If your option suits me, I would use it. Why are you suggesting something that doesn't fit?

**Reply by michalpulpan** [reply]
> @GabenGar  Yes, since the question says about "country/language", it means that the country should be in the url.  If your option suits me, I would use it. Why are you suggesting something that doesn't fit?

Same for me. I need country code in the URL in order to set correct currency and generally "country bias" info for my customers. 😕

**Reply by GabenGar** [reply]
@ermakov95 
But it does fit because BCP-47 can also include ISO_3166-1 codes.
@michalpulpan 
All of these should be account settings/cookies at best. And locale just translating them.
